File: raspberry-pi-mouse-ios/RaspberryPiMouseiOS/Data/Infrastructure/Ros/Topic/Subscribe/RosTopicUnsubscribe.swift
//
//  RosTopicUnsubscribe.swift
//  RaspberryPiMouseiOS
//
//  Created by Shusuke Ota on 2025/11/8.
//

import Foundation

/// [Unsubscriber Protocol](https://github.com/RobotWebTools/rosbridge_suite/blob/ros2/ROSBRIDGE_PROTOCOL.md#335-unsubscribe)
struct RosTopicUnsubscribe: RosTopicUnsubscribeProtocol {
    let id: String?
    let op: RosTopicOperation
    let topic: String

    enum CodingKeys: String, CodingKey {
        case id
        case op
        case topic
    }

    init(id: String? = nil, topic: String) {
        self.id = id
        self.op = RosTopicOperation.unsubscribe
        self.topic = topic
    }
}

File: raspberry-pi-android-client/app/src/main/java/com/shusuke/raspberry_pi_client/data/infrastructure/ros/topic/RosTopicUnsubscribe.kt
package com.shusuke.raspberry_pi_client.data.infrastructure.ros.topic

import com.shusuke.raspberry_pi_client.data.infrastructure.rosbridge.RosBridgeMessageOperation
import kotlinx.serialization.Serializable

/**
 * [Unsubscribe Protocol](https://github.com/RobotWebTools/rosbridge_suite/blob/ros2/ROSBRIDGE_PROTOCOL.md#335-unsubscribe)
 */
@Serializable
data class RosTopicUnsubscribe(
    val id: String? = null,
    val op: RosBridgeMessageOperation = RosBridgeMessageOperation.unsubscribe,
    val topic: String
)
